dhcpv4/nclient4/client.go

```diff
@@ -249,54 +303,57 @@ func (c *Client) receiveLoop() {
 }
 
 // ClientOpt is a function that configures the Client.
-type ClientOpt func(*Client)
+type ClientOpt func(c *Client) error
 
 // WithTimeout configures the retransmission timeout.
 //
 // Default is 5 seconds.
 func WithTimeout(d time.Duration) ClientOpt {
-	return func(c *Client) {
+	return func(c *Client) (err error) {
 		c.timeout = d
+		return
 	}
 }
 
-// WithSummaryLogger logs one-line DHCPv4 message summarys when sent & received.
-func WithSummaryLogger() ClientOpt {
-	return func(c *Client) {
-		c.logger = shortSummaryLogger{
-			Logger: log.New(os.Stderr, "[dhcpv4] ", log.LstdFlags),
-		}
+// WithLogger set the logger (see interface Logger).
+func WithLogger(newLogger Logger) ClientOpt {
+	return func(c *Client) (err error) {
+		c.logger = newLogger
+		return
 	}
 }
 
-// WithDebugLogger logs multi-line full DHCPv4 messages when sent & received.
-func WithDebugLogger() ClientOpt {
-	return func(c *Client) {
-		c.logger = debugLogger{
-			Logger: log.New(os.Stderr, "[dhcpv4] ", log.LstdFlags),
-		}
+// WithUnicast forces client to send messages as unicast frames.
+// By default client sends messages as broadcast frames even if server address is defined.
+func WithUnicast() ClientOpt {
```

Comment:
Can you allow users to specify a *net.UDPAddr here?

If they don't specify one, choose the default like you did here?